src/app.ts

```diff
@@ -13,14 +14,44 @@ interface TemperatureSummary {
   average: number
 }
 
+const temperatures: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (const reading of readings) {
+    temperatures.push(reading)
+  }
+  return
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const groupReadings = temperatures.reduce(
+    (tmp: any, obj: TemperatureReading) => {
```

Comment:
Try to avoid any types at all cost. 

Assuming that `temperatures` is typed correctly, you should not have to type `obj` as `TemperatureReading`. 

Also, try to have more descriptive names for these variables. `obj` and `tmp` are not super clear to me. I like to name the variables as close to as I can to what they are. For example in a map...

```
users.map(user => user.id)
```

A single item in the users array is a user. Try to do something like that here

Reply:
Yes, I will be more specific on the variable names

Reply:
Hello, I changed this on the newest push